lib/rex/proto/http/client.rb

```diff
@@ -23,7 +23,7 @@ class Client
   #
   # Creates a new client instance
   #
-  def initialize(host, port = 80, context = {}, ssl = nil, ssl_version = nil, proxies = nil, username = '', password = '')
+  def initialize(host, port = 80, context = {}, ssl = nil, ssl_version = nil, proxies = nil, username = '', password = '', comm: nil)
```

Comment:
When specified, the `comm` will be passed to the `Rex::Socket` API to be explicitly selected as the best comm for creating the socket. This bypasses the routing table and ensures the HTTP request is made through the specified session.